zuul/model.py

```diff
@@ -1431,8 +1431,19 @@ def matches(self, change):
                 return False
 
         if self.statuses:
-            if change.status not in self.statuses:
-                return False
+            # handle a change where status is list of statuses rather
+            # than a single string
+            if change.status and isinstance(change.status, str):
+                if change.status not in self.statuses:
+                    return False
+            elif isinstance(change.status, list):
+                # this is likely from github, where the head of the
+                # pr can have multiple statues on it.
+                # If the change statuses and the filter statuses are
+                # a null intersection, there are no matches and we return
+                # false.
+                if not set(change.status).isdisjoint(set(self.statuses)):
+                    return False
```

Comment:
I think the logic here is wrong.  isdisjoint() will return True when there is no intersection, so this'll incorrectly return False when there should be a match.  Its probably easier to test 'if not set(change.status).issubset(set(self.statuses))' instead?